Lib/_pyrepl/console.py

```diff
@@ -195,7 +196,19 @@ def runsource(self, source, filename="<input>", symbol="single"):
                 ast.PyCF_ONLY_AST,
                 incomplete_input=False,
             )
-        except (SyntaxError, OverflowError, ValueError):
+        except SyntaxError as e:
+            # If it looks like pip install was entered (a common beginner
+            # mistake), provide a hint to use the system command prompt.
+            if re.match(r"^\s*(py(thon3?)? -m pip|pip3?) install.*", source):
```

Comment:
```suggestion
            if re.match(r"^\s*(pip3?|py(thon3?)? -m pip) install.*", source):
```

~~Pretty sure that keeps the overall regex meaning the same, while making the optional suffix handling style more internally consistent.~~

Edit: changed the suggestion to flip the order of the branches, since my first suggestion of removing the pattern alternation entirely was wrong (as per the comments below)

Reply:
I think this would actually change the meaning of the regex, and that the updated version won't match a regular `pip install foo`, only ones that use the `python3 -m pip` form.  The `|` in the original regex isn't saying "`pip` or `pip3`" for that one little part, but rather that the whole parenthesized bit there must match "`(py(thon3?)? -m pip` or `pip3?`".

Reply:
Ah, you're right, I always forget how broad the scope of `|` is when mentally parsing regexes.

Reply:
Reworked the suggestion to instead be to put the simpler branch first. I don't think that's a merge blocker though.